### ISSUE TITLE
Certificate retrieval possible resource exhaustion

### ISSUE THREAD

**Issue by 0xDiBa** [mozilla/TLS-Observer#11]
Lots of error responses on certificate retrieval. Maybe due to file descriptor exhaustion. 
Possible fix with c3712e1b6ef3b05725637361838abc5fe5379fc5. Must check it out though.

**Comment by 0xDiBa** [comment]
func worker(msg []byte, ch *amqp.Channel) {

	  <-sem
	  defer releaseSemaphore()

	  certs, ip, err := tlsretriever.CheckHost(string(msg), "443", true) //<-sometimes code does not return 
	  panicIf(err)
	  if certs == nil {
	      log.Println("no certificate retrieved from", string(msg))
              return
	  }

After some debugging ( seeing that limiting the resources used does not solve the problem )
I observed random behavior regarding the actual certificate retrieval made. For example, using specific domains ( google.com and amazon.com ) and after several tests the calculated retrieval success rate was 1890-1995/2000 requests. No errors were printed from the checkhost function.

After some more debugging it seems that the code gets stuck here:

    conn, err := tls.Dial("tcp", canonicalName, &config)

in the tlsretriever package. 

Searching the TLS Go code I found that the default timeout for this connection is no timeout but it still 
doesn't make sense to me that it happens only a couple of times.
